.codacy.yml

```diff
@@ -0,0 +1,13 @@
+exclude_paths:
+  - '.travis/**'
+  - 'conf/**'
+  - 'doc/**'
```

Comment:
There are scripts and things in some of these dirs which we might not want to exclude from code checks.

When I extended pycodestyle to run over everything it found syntax errors and hidden code which shouldn't have been allowed to become neglected. I would be inclined to keep checkers (linters etc) as broad as possible.

If nothing else it would be good to check the python scripts in `doc`.

Reply:
Added back `doc`, skipping only the `joss-paper` folder. Also added back .travis folder. Now I think the only Python code not checked are some files under the `/tests/` folder, and the other test files.

So far that's the only option if we keep the plan to use PyTest and keep using Codacy, as Codacy will trigger one new warning for each `assert` used.

Reply:
There is another way of getting around the Pytest `assert` problem. https://github.com/metomi/rose/pull/2315

Reply:
https://github.com/metomi/rose/pull/2315/commits/d623ced4ea58446d6b8f917d8f5b33cbba170033

What about this @kinow?

Reply:
I had seen this one while looking for a fix for this problem @hjoliver 

The only issue I had with this solution, is that as far as I understand, that disables the bandit `B101` check in the project... which I think is not what we want.

We should not use `assert` in Cylc code for production... what we actually want is disable that verification for PyTest. If there's a way to apply that rule in Bandit for only a few folders, then I think that would be good too.

Otherwise I vote to check for `assert` in Cylc, except in the test folders.

Reply:
But I could be missing something here; I did not read the Bandit documentation to confirm my assumption :+1: happy to amend my change if necessary.

Reply:
> The only issue I had with this solution, is that as far as I understand, that disables the bandit B101 check in the project... which I think is not what we want.
> 
> We should not use assert in Cylc code for production..

OK, I buy that argument.  I imagine bandit can be told to ignore individual lines, with a special comment (`#nosec`?) ... but that would be painful for all the tests.

Let's leave it as is, unless @oliver-sanders comes back with more...